### PR TITLE
bpo-43106: Add os.O_EVTONLY/O_FSYNC/O_SYMLINK/O_NOFOLLOW_ANY

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue43106

### CODE REVIEW

File: Doc/library/os.rst [hunk 1]
```diff
@@ -1090,6 +1090,13 @@ or `the MSDN <https://msdn.microsoft.com/en-us/library/z0kc8e3z.aspx>`_ on Windo
 
    The above constants are only available on Windows.
 
+.. data:: O_EVTONLY
+          O_FSYNC
+   The above constants are only available on macOS.
```
Comment (serhiy-storchaka) [inline Doc/library/os.rst:1098]:
```suggestion

   The above constants are only available on macOS.
```

**Review by serhiy-storchaka — COMMENTED** [review]
What's New.

**Comment by corona10** [comment]
> What's New.

I intended to update What's New when we decide what flags will be added.
What do you think?

**Comment by serhiy-storchaka** [comment]
I think that all flags should be added. Even if they are not used in the stdlib, they can be used in third-party projects.

**Comment by corona10** [comment]
> I think that all flags should be added. Even if they are not used in the stdlib, they can be used in third-party projects.

Got it then I will add all flags through this PR and update the what's news

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -15301,6 +15304,9 @@ all_ins(PyObject *m)
 
 #if defined(__APPLE__)
     if (PyModule_AddIntConstant(m, "_COPYFILE_DATA", COPYFILE_DATA)) return -1;
+    if (PyModule_AddIntMacro(m, O_EVTONLY)) return -1;
+    if (PyModule_AddIntMacro(m, O_FSYNC)) return -1;
+    if (PyModule_AddIntMacro(m, O_SYMLINK)) return -1;
```
Comment (ronaldoussoren) [inline Modules/posixmodule.c]: Please use the same pattern as for the other O_* constants, outside of the ``#if __APPLE__`` block.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]

**Comment by corona10** [comment]
I have made the requested changes; please review again

**Review by ronaldoussoren — APPROVED** [review]
LGTM

**Pull request merged**